cocos/scene-graph/node.ts

```diff
@@ -2101,41 +2121,57 @@ export class Node extends CCObject implements ISchedulable, CustomSerializable {
         let childMat: Mat4;
         let childPos: Vec3;
         let dirtyBits = 0;
+        let positionDirty = 0;
+        let rotationScaleSkewDirty = 0;
+        let uiSkewComp: UISkew | null;
 
         while (i) {
             child = dirtyNodes[--i];
             childMat = child._mat;
             childPos = child._pos;
             dirtyBits |= child._transformFlags;
+            positionDirty = dirtyBits & TransformBit.POSITION;
+            rotationScaleSkewDirty = dirtyBits & TransformBit.RSS;
             if (cur) {
-                if (dirtyBits & TransformBit.POSITION) {
+                if (positionDirty && !rotationScaleSkewDirty) {
                     Vec3.transformMat4(childPos, child._lpos, cur._mat);
                     childMat.m12 = childPos.x;
                     childMat.m13 = childPos.y;
                     childMat.m14 = childPos.z;
                 }
-                if (dirtyBits & TransformBit.RS) {
+                if (rotationScaleSkewDirty) {
                     Mat4.fromSRT(childMat, child._lrot, child._lpos, child._lscale);
+                    uiSkewComp = child._uiProps._uiSkewComp;
```

Comment:
Don't use getter and make _uiSkewComp `@engineInternal` public property for fast accessing the UISkew component.